### PR TITLE
make composer view reusable, reply IMAP drafts, mailto IMAP drafts, reply attachment

### LINKED ISSUE #795: Unified composer view
ATM I know of at least 2 different composer views (+ different js view code for each of them), but we will need more of them:
- new message composer
- reply message composer
- forward message composer
- mailto-link message composer

To make things nice and easy we should combine all of them (or reuse the first one), so everywhere we use a composer, the same features are available (e.g. HTML content, drafts, attachment)

this is needed for at least the following issues #95, #602, #603, #605, #734, #735

### PR DESCRIPTION
**Pull request by ChristophWurst**
~~first step towards #795~~ this is #795 :rocket: 

~~My intention is to neither change nor break any functionality with this PR, I only want to make the new-message-composer view ready for using it as reply composer too (and then implement #602)~~
As this is much harder than I thought in the first place, I'll focus on not breaking the _new message composer_ functionality, other composers (e.g. mailto, reply) will be removed/replaced anyway.

Things to do:
- [x] init
- [x] reset
- [x] send mail
- [x] save IMAP draft
- [x] HTML classes/IDs
- [x] use backbone [$-selector](http://backbonejs.org/#View-dollar)
- [x] fix autocompletion
- [x] add draft UID when sending the message
- [x] hide subject when opening replies
- [x] send replies
- [x] create IMAP reply drafts
- [x] fix reply composer message body size
- [x] expand reply body on click
- [x] reply to/cc not filled correctly
- [x] submit button name is 'Send' for both new messages and replies
- [x] reply cc/bcc toggle slides down and up again (after opening cc/bcc in the new-message-composer)
- [x] fix attachment
- [x] test and fix mailto composer
- [x] add missing reply subject
- [ ] mailto composer does not show attachment names, ``{{displayName}}``is shown instead

fixes #795, #460, #602, #568

### CODE REVIEW

**Comment by jancborchardt** [comment]
Wohoo, major improvement. :rocket: :)

**Comment by ChristophWurst** [comment]
At last I finished this PR. This was more work to do than I expected, but no we have one composer for new messages, replies and the mailto compose window. All of them finally support drafts and attachments :smile: 

There's one known bug: the mailto composer does not show the filename of attached fiels, instead ``{{diplayName}}`` is shown. Could be a Backbone related issue, but I couldn't figure out how to fix it :-/

**Comment by jancborchardt** [comment]
Another bug: When deleting a message, it doesn’t switch to the next newest message. :) Works in master.

**Comment by ChristophWurst** [comment]
> Another bug: When deleting a message, it doesn’t switch to the next newest message. :) Works in master.

@jancborchardt yes, there's a bug. However, I could also reproduce this error on my production system (master branch):
bildschirmfoto von 2015-06-29 17-09-09

**Comment by jancborchardt** [comment]
Hm, works fine for me in master at least.

**Comment by ChristophWurst** [comment]
@jancborchardt should be fixed now ;-)

The TypeError still exists, but seems not to be related to this PR. Is thrown in FF but not in Chrome btw ;-)

File: js/views/composer.js [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+	folderId: null,
+	messageId: null,
+
+	draftInterval: 1500,
+	draftTimer: null,
+	draftUID: null,
+	hasData: false,
+	autosized: false,
+
+	events: {
```
Comment (DeepDiver1975) [inline js/views/composer.js]: Really nice! I wanted to get rid of all those jquery-fu in mail.js for long time! 

@ChristophWurst 

youre-awesome

**Comment by DeepDiver1975** [comment]
@ChristophWurst travis is crying because of this

````
js/send-mail.js: line 1, col 1, 'Mail' is defined but never used.

**Comment by DeepDiver1975** [comment]
I did not yet test this - but the code looks really great! Awesome job!

**Comment by ChristophWurst** [comment]
> @ChristophWurst travis is crying because of this
> 
> js/send-mail.js: line 1, col 1, 'Mail' is defined but never used.

I removed it, now travis should be happy again :-)

**Comment by DeepDiver1975** [comment]
let's fix the mailto composer in a second pr - this is awesome - time to move it in :+1:

**Pull request merged**